### ISSUE TITLE
Should unknown (undocumented?) services, applications or functionality be a top-level risk?

### ISSUE THREAD

**Issue by psiinon** [OWASP/Top10#63]
According to the OWASP Summit Top 10 Outcome: "Audience is everyone in appsec, and not just developers"
On this basis I think the Top 10 should include something like "Unknown services, applications or functionality" as a top level risk.
I have no evidence to support this, but I'm pretty sure this is a huge risk for most organizations - not knowing what services departments have set up, or not appreciating aspects of known services that introduce much more risk that was otherwise understood.

**Comment by cy4n** [comment]
clarify please: do you mean "unknown" as in "using dependencies or images you haven't checked out just because stackoverflow told me to"?

**Comment by psiinon** [comment]
Clarification - by 'unknown' I was thinking from the perspective of the security team - someone will know about these things, but in this case not the people who worry most about security.

**Comment by psiinon** [comment]
Twitter thread, for info: https://twitter.com/psiinon/status/874607268628713472

**Comment by jvehent** [comment]
I'm unsure if it is realistic to ask of a security team to know all services, applications or functionalities. In many organizations, that's just way too much information to keep up to date.

I think a security team should have up to date visibility on the technical and business risks of an organization. `Insufficient Risk Data` essentially means a security team is unable to focus on the most important issues because it has no visibility on what those are.

**Comment by Neil-Smithline** [comment]
@psiinon, what do you mean by "unknown"? If a random library goes through security scanning, does that make it known? 

_Sent from my Google Pixel XL using [FastHub](https://play.google.com/store/apps/details?id=com.fastaccess.github)_

**Comment by sslHello** [comment]
@psiinon: Hi Simon, is it about missing structured documentation? Something like ICT Asset Management for assets like applications including their business impacts and their logical and physical components? (=Something more than a CMDB)? 
Do you think, that it could be a solution to extend "O - What's Next for Organizations" to integrate this issue? (And perhaps a tiny more technical part of it in "A5-Security Misconfiguration" if it happens even without knowledge of the responsible application manager?)  
Top-10-2013 'Risk/What's My Risk?' added the business specific impact of a risk caused by an application, which could be seen as a first step to this complex but very interesting topic. Perhaps this shoud be extended, too.
Thank you for you nice comment!

**Comment by psiinon** [comment]
@Neil-Smithline I was thinking of applications / functionality etc that the teams / people responsible for securing the organization are not aware of. If they dont know dept X has put up a website backed by an old an insecure version of WordPress then they cant take steps to secure it.

**Comment by psiinon** [comment]
@sslHello its definitely an asset management problem. Could be missing documentation. I just the impression that many businesses really dont know the full extent of the applications that make up their attack surface area. What you dont know about you cant secure.

**Comment by Neil-Smithline** [comment]
Do you think that this could be part of "A9 – Using Components with Known Vulnerabilites"?

**Comment by psiinon** [comment]
yes, I think that could be expanded to include this.

**Comment by davewichers** [comment]
And unknown capability  is very different in my opinion from use of a known
vulnerable component. I wouldn't recommend trying to shoehorn that topic
into A9.

**Comment by sslHello** [comment]
Hi Simon.

I've just started to include some of your suggestions. I've added the following text to '+O: Get Started': 
* Document all your IT assets (e.g. applications) in a Configuration Management Database (CMDB).

And rephrased the 'Risk Based Portfolio Approach':
* Identify the protection needs of your application portfolio from a business perspective. Add the results to your CMDB. 
* Establish a common risk rating model with a consistent set of likelihood and impact factors reflective of your organization's tolerance for risk.
* Accordingly measure and prioritize all your applications and APIs. Add the results to your CMDB. 
* Establish assurance guidelines to properly define coverage and level of rigor required.

I plan to add a new page 'What’s Next for **Product Onwers** or **Application Managers**'. If this works, I'd like to get more detailed, e.g. 'to document all services and interfaces of an application'. 

Thanks a lot.
Cheers Torsten
